controllers/datadogagent/utils.go

```diff
@@ -1153,6 +1153,23 @@ func getVolumeMountsForAgent(spec *datadoghqv1alpha1.DatadogAgentSpec) []corev1.
 			},
 		}...)
 	}
+
+	// SystemProbe volumes
+	if datadoghqv1alpha1.BoolValue(spec.Agent.SystemProbe.Enabled) {
+		volumeMounts = append(volumeMounts, []corev1.VolumeMount{
+			{
+				Name:      datadoghqv1alpha1.SystemProbeSocketVolumeName,
+				MountPath: datadoghqv1alpha1.SystemProbeSocketVolumePath,
+				ReadOnly:  true,
+			},
+			{
+				Name:      datadoghqv1alpha1.SystemProbeConfigVolumeName,
+				MountPath: datadoghqv1alpha1.SystemProbeConfigVolumePath,
+				SubPath:   datadoghqv1alpha1.SystemProbeConfigVolumeSubPath,
+			},
```

Comment:
Can the config file be mounted read-only ?

```suggestion
			{
				Name:      datadoghqv1alpha1.SystemProbeConfigVolumeName,
				MountPath: datadoghqv1alpha1.SystemProbeConfigVolumePath,
				SubPath:   datadoghqv1alpha1.SystemProbeConfigVolumeSubPath,
				ReadOnly:  true,
			},
```